extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/BridgeMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,85 @@
+
+import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.config.Config;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.DsAPI;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.impl.DsAPIImpl;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryService;
+
+/**
+ * The {@link BridgeMDNSDiscoveryParticipant} is responsible for discovering new and
+ * removed digitalSTROM bridges. It uses the central {@link MDNSDiscoveryService}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ *
+ */
+public class BridgeMDNSDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    // private org.slf4j.Logger logger = LoggerFactory.getLogger(DssBridgeMDNSDiscoveryParticipant.class);
+
+    private String hostAdress = null;
+    private String dsid = null;
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(DigitalSTROMBindingConstants.THING_TYPE_DSS_BRIDGE);
+    }
+
+    @Override
+    public String getServiceType() {
+        return "_tcp.local.";
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo service) {
+        ThingUID uid = getThingUID(service);
+
+        Map<String, Object> properties = new HashMap<>(2);
+        properties.put(DigitalSTROMBindingConstants.HOST, hostAdress);
+        properties.put(DigitalSTROMBindingConstants.DS_ID, dsid);
+        if (uid != null) {
+            return DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel("DigitalSTROM-Server")
+                    .build();
+        }
+        return null;
+    }
+
+    @Override
+    public ThingUID getThingUID(ServiceInfo service) {
+        // logger.debug("URL: " + service.getName() + "." + service.getDomain() + ".");
+        if (service.getApplication().contains("dssweb")) {
+            hostAdress = service.getName() + "." + service.getDomain() + ".";
+            DsAPI digitalSTROMClient = new DsAPIImpl(hostAdress, Config.DEFAULT_CONNECTION_TIMEOUT,
+                    Config.DEFAULT_READ_TIMEOUT);
+            dsid = digitalSTROMClient.getDSID("123");
+            if (dsid != null) {
+                return new ThingUID(DigitalSTROMBindingConstants.THING_TYPE_DSS_BRIDGE, dsid);
+            } else {
+                // The dSID can only read out if the authentication is set, so we give the number 1
+                return new ThingUID(DigitalSTROMBindingConstants.THING_TYPE_DSS_BRIDGE, "1");
```

Comment:
??? You will end up with the same ThingUID for all discovered dSS - this simply does not work.
Doesn't the ServiceInfo provide ANY unique information that could be used?